go.mod

```diff
@@ -59,7 +59,7 @@ require (
 	go.uber.org/zap v1.27.0
 	golang.org/x/crypto v0.30.0
 	golang.org/x/exp v0.0.0-20230522175609-2e198f4a06a1
-	golang.org/x/mod v0.22.0
+	golang.org/x/mod v0.20.1-0.20240815161730-b1d336cfca97 // this version is equivalent to 0.22.0 except it excludes a commit that enforces go directive patch version be used https://github.com/golang/mod/commit/3afcd4e90a74c23515a9543f1e8fb68f05ecc8e0 We would want to remove this when 0.23.0 is released
```

Comment:
@julio-lopez Is this ok?

Reply:
> on Go versions 1.21.0 up to 1.21.10,
> the toolchain upgrade logic will try to download the release "1.22",
> which doesn't exist.

The patch difference with 0.22.0 do not applies beneficially to this repo which is already building on 1.22+ go binaries.

Reply:
I would expect by 1.23.0 go if golang/mod upgrades to it that they would drop `1.22.0` to `1.23` instead of `1.23.0`

Reply:
long term it might not make sense to use this module, it's only used for the semver package at the moment.

This module states in its [readme](https://github.com/golang/mod)
> packages for writing tools that work directly with Go module mechanics

Reply:
I've been told the alternative to excluding this is to set toolchain directive to 1.22.10 which should not affect imported modules.
go directive can remain at 1.22.0.

Reply:
I have additionally requested changes at https://github.com/kopia/kopia/pull/4248#pullrequestreview-2518093777 to satisfy the same requirement.